src/components/DataTable/DataTable.tsx

```diff
@@ -52,19 +54,26 @@ const extractIconFromCell = (cellValue: unknown): React.ReactElement | undefined
 export const MoonstoneTable = <T extends Record<string, unknown>>({
     data,
     columns,
-    isStructured = false
+    isStructured = false,
+    enableRowSelection = false
 }: DataTableProps<T>) => {
     const [expanded, setExpanded] = useState<ExpandedState>({});
+    const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
```

Comment:
We also need to add a `defaultSelection` prop to automatically select rows when the table loads.